### DISCUSSION TITLE
snmp options

### DISCUSSION THREAD

**Discussion by optimuscream** [akvorado/akvorado#37] [Q&A] [answered]
Hi

Great project.
I have real flow I want to monitor. The problem is where to put the definition of the flow in the configuration ? Is in the inlet ?
I can see the flow with tcpdump and also the akvorado accept it after I change the snmp to our snmp community in

```
snmp:
  workers: 10
  defaultcommunity: ourcommunity

```
Thanks and regards

**Comment by optimuscream** [comment]
![Screenshot 2022-07-26 at 08-41-58 Akvorado Home](https://user-images.githubusercontent.com/14342926/180904387-bb252a59-7a07-410d-abd5-51500050f00d.png)

I have make it work with real flow data. The problem now comes from GeoIP database. Can not see TopAS and Top Countries.  I have install geoip-database package (Debian 11) and installed in /usr/share/GeoIP/   . Also I have register and put the ID and Key in the config for GeoIP Lite Service from Maxmind.

Maybe i missed something here, need advice.

Thank you

**Reply by vincentbernat** [reply]
Use `docker-compose logs geoip` to check if the container is able to get the geoip databases from Maxmind. If yes, restart the inlet container as it is configured to not use geoip if the databases were not present at start.

**Reply by optimuscream** [reply]
```
docker-compose logs geoip

Attaching to akvorado_geoip_1
geoip_1                     | # STATE: Creating configuration file at /etc/GeoIP.conf
geoip_1                     | # STATE: Running geoipupdate
geoip_1                     | # STATE: Sleeping for 48 hours
geoip_1                     | Terminated
geoip_1                     | # STATE: Creating configuration file at /etc/GeoIP.conf
geoip_1                     | # STATE: Running geoipupdate
geoip_1                     | # STATE: Sleeping for 48 hours
geoip_1                     | Terminated
```
at first no /etc/GeoIP.conf .. after I install the package geoipupdate , the file created . I put the account ID and key accordingly.

```
AccountID myid
LicenseKey mylicensekey
EditionIDs GeoLite2-Country GeoLite2-ASN

```
But still not updating.
I have no idea where to look .

Thanks.

**Reply by vincentbernat** [reply]
The file is inside the container. Don't create it. Dunno why it says "Terminated" but no other error. The key needs to be put in `docker-compose.yml`, the `docker-compose up -d geoip`, then check again `docker-compose logs geoip`.

**Reply by optimuscream** [reply]
> The file is inside the container. Don't create it. Dunno why it says "Terminated" but no other error. The key needs to be put in `docker-compose.yml`, the `docker-compose up -d geoip`, then check again `docker-compose logs geoip`.

The terminated line was from stopping and starting the docker-image .
ok I understand . I mixed it up with the host of the docker.
But where should I put the definition of /etc/GeoIP.conf in the docker-compose.yml file ? 

Thanks.

**Reply by vincentbernat** [reply]
You don't, it's created by the container on the fly, with the information you provide in the geoip section. This part seems to work. Be sure to restart the inlet component and it should work. Otherwise, check its logs (`docker-compose logs akvorado-inlet | grep -i geoip`)

**Reply by optimuscream** [reply]
> You don't, it's created by the container on the fly, with the information you provide in the geoip section. This part seems to work. Be sure to restart the inlet component and it should work. Otherwise, check its logs (`docker-compose logs akvorado-inlet | grep -i geoip`)


Ok ..
Yes ..already do that .. 

docker-compose logs akvorado-inlet | grep -i geoip

last line show :

akvorado-inlet_1            | {"level":"info","time":"2022-07-26T08:02:51Z","caller":"akvorado/inlet/geoip/root.go:104","module":"akvorado/inlet/geoip","message":"starting GeoIP component"}

and not stopping.

Can I ls or cat the file inside the container ? 

I still not understand why Top AS and Top Country not show anything. Maybe Maxmind don't have or not updated our ASN / Country in their database. I'll have to check it.

Thanks .

**Reply by vincentbernat** [reply]
You can use `docker-compose exec` to execute commands inside the container.

**Comment by optimuscream** [comment, marked as answer]
The geoip image base on Alpine Linux .. and yes its all there. 

```
docker-compose exec geoip cat /etc/os-release
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.15.0
PRETTY_NAME="Alpine Linux v3.15"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://bugs.alpinelinux.org/"

docker-compose exec geoip ls /etc/GeoIP.conf
/etc/GeoIP.conf
and all the config is there and true.

docker-compose exec geoip ls /usr/share/GeoIP
GeoLite2-ASN.mmdb      GeoLite2-Country.mmdb
```

In real os I can check and read IP to AS/Country mapping with mmdbinspect from maxmind tools from :

https://github.com/maxmind/mmdbinspect

```
mmdbinspect --db /usr/share/GeoIP/GeoLite2-ASN.mmdb 202.158.3.7/32
[
    {
        "Database": "/usr/share/GeoIP/GeoLite2-ASN.mmdb",
        "Records": [
            {
                "Network": "202.158.3.7/17",
                "Record": {
                    "autonomous_system_number": 4787,
                    "autonomous_system_organization": "PT Cyberindo Aditama"
                }
            }
        ],
        "Lookup": "202.158.3.7/32"
    }
]
```

how do I install or modify the image to inject the binary in /usr/bin .. or could you do it ?
I want to debug.

Or should I make another modification to akvorado.yaml or docker-compose.yml to solve the issue.

Sorry for the noise , this all new to me and very exciting project.

Thank you

**Reply by vincentbernat** [reply]
Check the `inlet_geoip` metrics. You should get something like that:
```
akvorado_inlet_geoip_db_refresh_total{database="asn"} 3
akvorado_inlet_geoip_db_refresh_total{database="country"} 3
```

**Reply by optimuscream** [reply]
Where to look or what command should I do ? Sorry no idea where to look / what to do ..

**Reply by vincentbernat** [reply]
Look at the troubleshooting documentation, there are commands to check the metrics.

**Reply by optimuscream** [reply]
I see. ok.

```
curl -s http://akvorado:8081/api/v0/inlet/metrics | grep inlet_geoip
# HELP akvorado_inlet_geoip_db_refresh_total Refresh event for a GeoIP database.
# TYPE akvorado_inlet_geoip_db_refresh_total counter
akvorado_inlet_geoip_db_refresh_total{database="asn"} 1
akvorado_inlet_geoip_db_refresh_total{database="country"} 1
```
Other type of troubleshooting looks ok . Flow ok,

```
curl -s http://akvorado:8081/api/v0/inlet/flows\?limit=1
curl -s http://akvorado:8081/api/v0/inlet/metrics | grep '^akvorado_inlet'
```

show good result. updated every 10 seconds with different flow information.

`curl -s http://akvorado:8081/api/v0/inlet/metrics | grep '^akvorado_inlet_flow_input_udp_packets'`

packet counting

`curl -s http://akvorado:8081/api/v0/inlet/metrics | grep '^akvorado_inlet_core_flows_forwarded'`

packet forwarded

`curl -s http://akvorado:8081/api/v0/inlet/metrics | grep '^akvorado_inlet' | grep _errors`

Error not increase. Error number stop

Thanks.

**Reply by georgetasioulis** [reply]
When you installed the `geoip-database package` earlier to /usr/share/GeoIP/  as you mentioned in https://github.com/vincentbernat/akvorado/discussions/37#discussioncomment-3227117 did you delete it afterwards so that the docker instance can mount that directory?

Which release version are you running?

I'm exporting flows from a Mikrotik edge router (ROS6) and AS/country lookups work fine. I'm running v1.5.3 btw
<img width="1205" alt="Screenshot 2022-07-28 at 11 40 34 AM" src="https://user-images.githubusercontent.com/460954/181462162-b8b4fc66-cb55-4710-aa55-46d4186a3ada.png">

All I did was add my `GEOIPUPDATE_ACCOUNT_ID` and `GEOIPUPDATE_LICENSE_KEY` in `docker-compose.yml`

**Reply by vincentbernat** [reply]
For me, it looks like this is working. Having `/usr/share/GeoIP` on the host is not a problem (the container is using its own file system where the directory does not exist and Docker is mounting the volume on it). Check the you have the same files in both container:
```
docker-compose exec geoip ls -l /usr/share/GeoIP
docker-compose exec akvorado-inlet ls -l /usr/share/GeoIP
```

**Reply by optimuscream** [reply]
Yes already inform this in https://github.com/vincentbernat/akvorado/discussions/37#discussioncomment-3256867

```
docker-compose exec geoip ls -l /usr/share/GeoIP
total 13060
-rw-r--r--    1 root     root       7809574 Jul 26 01:26 GeoLite2-ASN.mmdb
-rw-r--r--    1 root     root       5560649 Jul 26 01:26 GeoLite2-Country.mmdb

docker-compose exec akvorado-inlet ls -l /usr/share/GeoIP
total 13060
-rw-r--r--    1 root     root       7809574 Jul 26 01:26 GeoLite2-ASN.mmdb
-rw-r--r--    1 root     root       5560649 Jul 26 01:26 GeoLite2-Country.mmdb

docker-compose exec geoip cat /etc/GeoIP.conf
AccountID myaccountid
LicenseKey mylicensekey
EditionIDs GeoLite2-ASN GeoLite2-Country
```
but no /etc/GeoIP.conf in akvorado-inlet image.

I made changes in akvorado.yaml  in 
  asns:
    64501: My Company Name
  exporter-classifiers:
      -ClassifyRegion("indonesia")
      - ClassifyTenant("cbn")

to suit my company and region. I don't think that the source of the issue.

Thanks

**Reply by vincentbernat** [reply]
I have added counters. Once the [build](https://github.com/vincentbernat/akvorado/actions/runs/2753097467) is complete, you can use `ghcr.io/vincentbernat/akvorado:main` (instead of `:latest`) as an image to test what happens. We'll see if the counters are increasing and which ones are.

**Reply by optimuscream** [reply]
> I have added counters. Once the [build](https://github.com/vincentbernat/akvorado/actions/runs/2753097467) is complete, you can use `ghcr.io/vincentbernat/akvorado:main` (instead of `:latest`) as an image to test what happens. We'll see if the counters are increasing and which ones are.

Please inform when it's ready .

Thanks.

**Reply by optimuscream** [reply]
![image](https://user-images.githubusercontent.com/14342926/182525602-5faad653-53a3-492f-bf75-b7195578b4e1.png)

I still have problem with top source country and asn , but I have another thing that confuse me also . Why I have 21 exporters show on the dashboard above. If I check with command 

```
curl -s http://127.0.0.1:8080/api/v0/inlet/metrics | grep 'forward'
# HELP akvorado_inlet_core_flows_forwarded Number of flows forwarded to Kafka.
# TYPE akvorado_inlet_core_flows_forwarded counter
akvorado_inlet_core_flows_forwarded{exporter="xxx.100.5.80"} 429198
akvorado_inlet_core_flows_forwarded{exporter="xxx.100.5.81"} 489019

curl -s http://akvorado:8081/api/v0/inlet/metrics | grep -i akvorado_inlet_snmp_cache_exporters
# HELP akvorado_inlet_snmp_cache_exporters Number of exporters in cache.
# TYPE akvorado_inlet_snmp_cache_exporters gauge
akvorado_inlet_snmp_cache_exporters 2

```
There are only two forwarders / exporters .

For geoip case , this command show 

```
curl -s http://akvorado:8081/api/v0/inlet/metrics | grep -i geoip

# HELP akvorado_inlet_geoip_db_hits_total Number of hits for a GeoIP database.
# TYPE akvorado_inlet_geoip_db_hits_total counter
akvorado_inlet_geoip_db_hits_total{database="country"} 2.201188e+06
# HELP akvorado_inlet_geoip_db_misses_total Number of misses for a GeoIP database.
# TYPE akvorado_inlet_geoip_db_misses_total counter
akvorado_inlet_geoip_db_misses_total{database="country"} 13466
# HELP akvorado_inlet_geoip_db_refresh_total Refresh event for a GeoIP database.
# TYPE akvorado_inlet_geoip_db_refresh_total counter
akvorado_inlet_geoip_db_refresh_total{database="asn"} 1
akvorado_inlet_geoip_db_refresh_total{database="country"} 1

docker-compose logs geoip
Attaching to akvorado_geoip_1
geoip_1                     | # STATE: Creating configuration file at /etc/GeoIP.conf
geoip_1                     | # STATE: Running geoipupdate
geoip_1                     | # STATE: Sleeping for 48 hours

docker-compose exec geoip ls /etc/GeoIP.conf
/etc/GeoIP.conf
```

curl -s http://akvorado:8081/api/v0/inlet/flows\?limit=1 <-- shows updated flow each time I run the command

I'm using the main image

```
  akvorado-service: &akvorado-image
    # build: {context: .}
    image: ghcr.io/vincentbernat/akvorado:main
    command: version

```
Thanks

**Reply by vincentbernat** [reply]
The number of exporters include past and present exporters. As for geoip, it seems to work. The widget on the home page requires interface classification to work (it filter on InIfBoundary = external). Use the visualize tab without a filter and group by SrcCountry to check if geoip works.

**Reply by optimuscream** [reply]
> The number of exporters include past and present exporters. As for geoip, it seems to work. The widget on the home page requires interface classification to work (it filter on InIfBoundary = external). Use the visualize tab without a filter and group by SrcCountry to check if geoip works.

Yes it works ! .. SrcAS also works. I knew there's nothing wrong . 

But how to show the TopAS and TopCountry in pie chart in  the home tab ?? Where should I modify ?

Thanks.

**Reply by optimuscream** [reply]
> > The number of exporters include past and present exporters. As for geoip, it seems to work. The widget on the home page requires interface classification to work (it filter on InIfBoundary = external). Use the visualize tab without a filter and group by SrcCountry to check if geoip works.
> 
> Yes it works ! .. SrcAS also works. I knew there's nothing wrong .
> 
> But how to show the TopAS and TopCountry in pie chart in the home tab ?? Where should I modify ?
> 
> Thanks.

I know how to modify it . It works now. Thanks

**Reply by optimuscream** [reply]
> The number of exporters include past and present exporters. As for geoip, it seems to work. The widget on the home page requires interface classification to work (it filter on InIfBoundary = external). Use the visualize tab without a filter and group by SrcCountry to check if geoip works.

How do I clean / refresh the exporters , to reflect the real exporters in the dashboard ?

Thanks

**Reply by umiseaz** [reply]
> > > The number of exporters include past and present exporters. As for geoip, it seems to work. The widget on the home page requires interface classification to work (it filter on InIfBoundary = external). Use the visualize tab without a filter and group by SrcCountry to check if geoip works.
> > 
> > 
> > Yes it works ! .. SrcAS also works. I knew there's nothing wrong .
> > But how to show the TopAS and TopCountry in pie chart in the home tab ?? Where should I modify ?
> > Thanks.
> 
> I know how to modify it . It works now. Thanks

Can i know where should i modify ? for the top source as ?

**Reply by umiseaz** [reply]
> > > > The number of exporters include past and present exporters. As for geoip, it seems to work. The widget on the home page requires interface classification to work (it filter on InIfBoundary = external). Use the visualize tab without a filter and group by SrcCountry to check if geoip works.
> > > 
> > > 
> > > Yes it works ! .. SrcAS also works. I knew there's nothing wrong .
> > > But how to show the TopAS and TopCountry in pie chart in the home tab ?? Where should I modify ?
> > > Thanks.
> > 
> > 
> > I know how to modify it . It works now. Thanks
> 
> Can i know where should i modify ? for the top source as ?

ok i figured it out 
changed to:
  interface-classifiers:
    # This is an example. This must be customized depending on the
    # descriptions of your interfaces. In the following, we assume
    # external interfaces are named "Transit: Cogent" Or "IX:
    # FranceIX".
    - |
      ClassifyConnectivityRegex(Interface.Description, "^(?i)(transit|pni|ppni|ix):? ", "$1") &&
      ClassifyProviderRegex(Interface.Description, "^\\S+?\\s(\\S+)", "$1") &&
      ClassifyInternal()
    - ClassifyExternal()

**Comment by Kashif-Mehmood0** [comment]
Hi everyone,
I have installed Akvorado on Ubuntu-20 using Docker and removed the demo using the provided link.  

https://github.com/akvorado/akvorado/discussions/1649

However, I'm facing an issue — no data is showing on the UI.
I think I need to make some configuration changes in the YAML file. Can anyone please guide me on what exact changes I should make to get data to appear on the dashboard?

Thanks!

**Reply by roopeshsn** [reply]
The next step is to configure SNMP for polling interface names. 

Refer here: https://demo.akvorado.net/docs/configuration#metadata

This slide deck which is presented on FRnOG#36, will be useful for reference: https://vincent.bernat.ch/en/blog/2022-frnog36-akvorado

**Reply by Kashif-Mehmood0** [reply]
Can you please share a sample SNMP configuration on the Akvorado side? Below is my scenario:

I am exporting NetFlow data from a Cisco router running on EVE-NG. The Akvorado collector is hosted on a VMware virtual machine. Both the router and the Akvorado server are reachable over the network.

The router is configured to export flows using NetFlow v9 on UDP port 2055 to the Akvorado collector at 192.168.220.194.

On the Akvorado side, I only commented out the demo section in /akvorado/.env and config/akvorado.yaml.

Please find the full router configuration below:

Router#sh run
Building configuration...

Current configuration : 1833 bytes
!
! Last configuration change at 15:29:29 UTC Wed Jul 30 2025
!
version 15.5
service config
service timestamps debug datetime msec
service timestamps log datetime msec
no service password-encryption
!
hostname Router
!
boot-start-marker
boot-end-marker
!
!
!
no aaa new-model
!
!
!
bsd-client server url https://cloudsso.cisco.com/as/token.oauth2
mmi polling-interval 60
no mmi auto-configure
no mmi pvc
mmi snmp-timeout 180
!
!
!
!
!
!
!
!
!
!
!
!
!

!
!
ip dhcp pool LAN
network 172.16.0.0 255.255.255.0
dns-server 8.8.8.8
default-router 172.16.0.1
!
!
!
ip flow-cache timeout active 1
ip cef
no ipv6 cef
!
!
flow exporter AKVORADO
destination 192.168.220.194
transport udp 2055
!
!
flow monitor AKVORADO-MON
exporter AKVORADO
cache timeout active 60
record netflow-original
!
multilink bundle-name authenticated
!
!
!
!
!
!
!
cts logging verbose
!
!
!
redundancy
!
!
!
!
!
!
!
!
!
!
!
!
!
!
!
interface Ethernet0/0
ip address 192.168.220.200 255.255.255.0
ip flow monitor AKVORADO-MON input
ip flow monitor AKVORADO-MON output
ip flow ingress
ip flow egress
!
interface Ethernet0/1
ip address 172.16.0.1 255.255.255.0
ip flow monitor AKVORADO-MON input
ip flow monitor AKVORADO-MON output
ip flow ingress
ip flow egress
!
interface Ethernet0/2
no ip address
shutdown
!
interface Ethernet0/3
no ip address
shutdown
!
ip forward-protocol nd
!
ip flow-export version 9
ip flow-export template timeout-rate 1
ip flow-export template refresh-rate 1
ip flow-export destination 192.168.220.194 2055
!
no ip http server
no ip http secure-server
ip route 0.0.0.0 0.0.0.0 192.168.220.2
!
!
snmp-server community public RO
snmp-server ifindex persist
!
!
control-plane
!
!
!
!
!
!
!
!
line con 0
logging synchronous
line aux 0
line vty 0 4
login
transport input none
!
ntp server pnpntpserver.localdomain
!
end

Router#
